# HG changeset patch
# User sergey_azarovskyy <sazarovsky@adfsolutions.com>
# Date 1492857237 -10800
#      Sat Apr 22 13:33:57 2017 +0300
# Node ID f90d62d501f2171b66c8d7378a391fc394df2230
# Parent  338739aa303a72a1ea9ba40756f901a890b687ad
TGF-8171 crash in rar.patch

diff --git a/CPP/7zip/Compress/Rar2Decoder.cpp b/CPP/7zip/Compress/Rar2Decoder.cpp
--- a/CPP/7zip/Compress/Rar2Decoder.cpp
+++ b/CPP/7zip/Compress/Rar2Decoder.cpp
@@ -254,6 +254,9 @@
     }
     else if (sym >= kMatchNumber)
     {
+	  if (sym == 0xFFFFFFFF)
+		return false;  
+	  
       sym -= kMatchNumber;
       length = kNormalMatchMinLen + UInt32(kLenStart[sym]) +
         m_InBitStream.ReadBits(kLenDirectBits[sym]);
